File: TaeJin/Ex04/app/src/main/java/com/example/ex04/MainActivity.java
package com.example.ex04;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    Button btn,btn2;
    EditText edit;
    TextView text;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btn= (Button)findViewById(R.id.send);
        edit=(EditText)findViewById(R.id.edit);
        text=(TextView)findViewById(R.id.text);
        btn2=(Button)findViewById(R.id.close) ;
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String msg =edit.getText().toString();
                Toast.makeText(getApplicationContext()," 전송한다",Toast.LENGTH_LONG).show();
            }
        });

        btn2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });

        edit.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int i, int i1, int i2) {

            }

            @Override
            public void onTextChanged(CharSequence s, int i, int i1, int i2) {
                String input= edit.getText().toString();
                text.setText(input.length()+"/80글자");
            }

            @Override
            public void afterTextChanged(Editable editable) {

            }
        });


    }


}

File: 2024+/chapter2(counter)/app/src/main/java/com/example/chapter2counter/MainActivity.kt
package com.example.chapter2counter

import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val numberTextView = findViewById<TextView>(R.id.numberTextView)
        val resetButton = findViewById<Button>(R.id.resetButton)
        val plusButton = findViewById<Button>(R.id.plusButton)

        var number = 0

        resetButton.setOnClickListener {
            number = 0
            numberTextView.text = number.toString()
            Log.d("onClick","리셋 버튼 클릭")
        }

        plusButton.setOnClickListener {
            number++
            numberTextView.text = number.toString()
            Log.d("onClick","플러스 버튼 클릭")
        }
    }
}